packages/http.go

```diff
@@ -15,11 +15,34 @@ import (
 	"github.com/elastic/package-registry/util"
 )
 
-func ServePackage(w http.ResponseWriter, r *http.Request, p *Package) {
+// ServePackage is used by artifactsHandler to serve packages and signatures.
+func ServePackage(w http.ResponseWriter, r *http.Request, p *Package, packagePath string) {
+	span, _ := apm.StartSpan(r.Context(), "ServePackage", "app")
+	defer span.End()
+
+	if p.RemoteResolver() != nil {
+		p.RemoteResolver().RedirectArtifactsHandler(w, r, p)
+		return
+	}
+	serveLocalPackage(w, r, p, packagePath)
+}
+
+// ServePackageSignature is used by signaturesHandler to serve signatures.
+func ServePackageSignature(w http.ResponseWriter, r *http.Request, p *Package, packagePath string) {
+	span, _ := apm.StartSpan(r.Context(), "ServePackageSignature", "app")
+	defer span.End()
+
+	if p.RemoteResolver() != nil {
+		p.RemoteResolver().RedirectSignaturesHandler(w, r, p)
+		return
+	}
+	serveLocalPackage(w, r, p, packagePath)
+}
```

Comment:
Following with https://github.com/elastic/package-registry/pull/823#discussion_r907093896, good point about reusing the code for both things, but we could still get the paths from the package, do we don't need to pass both things.

```suggestion
func ServePackage(w http.ResponseWriter, r *http.Request, p *Package) {
	span, _ := apm.StartSpan(r.Context(), "ServePackage", "app")
	defer span.End()

	if p.RemoteResolver() != nil {
		p.RemoteResolver().RedirectArtifactsHandler(w, r, p)
		return
	}
	serveLocalPackage(w, r, p, p.BasePath)
}

// ServePackageSignature is used by signaturesHandler to serve signatures.
func ServePackageSignature(w http.ResponseWriter, r *http.Request, p *Package) {
	span, _ := apm.StartSpan(r.Context(), "ServePackageSignature", "app")
	defer span.End()

	if p.RemoteResolver() != nil {
		p.RemoteResolver().RedirectSignaturesHandler(w, r, p)
		return
	}
	serveLocalPackage(w, r, p, p.BasePath+".sig")
}
```